### PR TITLE
stats(native): fixes to align with wasm stats

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Fix an error in parsing in Wasm stats for endpoint encoding (parts == 4 leads to crash).
Adds fallbacks to native stats for endpoint encoding, blackhole/passthrough.
Handle empty values properly (should be unknown).
Finish CEL expression support.

### CODE REVIEW

File: testdata/stats/client_config_customized.yaml.tmpl [hunk 1]
```diff
@@ -1,8 +1,5 @@
 field_separator: ";.;"
 definitions:
-- name: requests_total
```
Comment (lei-tang) [inline testdata/stats/client_config_customized.yaml.tmpl:3]: Why is requests_total deleted?

Comment (kyessenov) [reply testdata/stats/client_config_customized.yaml.tmpl:3]: This test case was changing the value for a standard metric. I don't think we want to support this feature.

**Review by lei-tang — APPROVED** [review]

**Pull request merged**